### ISSUE TITLE
Error during Deployment: unknown Flag: --k8s-cluster-name

### ISSUE THREAD

**Issue by pickmylight** [kedacore/charts#583]
The Deployment of Keda with version 2.12.1 ist currently failing. And will be failing until fixes will be made. The error is a result of different patching mechanisms of the helm chart and the official 2.12.1 release. 

In Version 2.12.1 there is no option to set the `--k8s-cluster-name` flag whereas in the matching Helm Chart Version the Flag is set permamently.

## Expected Behavior

Keda runs the correct Version 2.12.1 with the right flags

## Actual Behavior

```
2023/12/05 11:40:36 maxprocs: Updating GOMAXPROCS=1: using minimum allowed GOMAXPROCS
unknown flag: --k8s-cluster-name
Usage of /keda:
      --cert-dir string                       Webhook certificates dir to use. Defaults to /certs (default "/certs")
      --cert-secret-name string               KEDA certificates secret name. Defaults to kedaorg-certs (default "kedaorg-certs")
      --disable-compression                   Disable response compression for k8s restAPI in client-go.  (default true)
      --enable-cert-rotation                  enable automatic generation and rotation of TLS certificates/keys
      --enable-opentelemetry-metrics          Enable the opentelemetry metric of keda-operator.
      --enable-prometheus-metrics             Enable the prometheus metric of keda-operator. (default true)
      --health-probe-bind-address string      The address the probe endpoint binds to. (default ":8081")
      --k8s-cluster-domain string             Kubernetes cluster domain. Defaults to cluster.local (default "cluster.local")
      --kube-api-burst int                    Set the burst for throttling requests sent to the apiserver (default 30)
      --kube-api-qps float32                  Set the QPS rate for throttling requests sent to the apiserver (default 20)
      --kubeconfig string                     Paths to a kubeconfig. Only required if out-of-cluster.
      --leader-elect                          Enable leader election for controller manager. Enabling this will ensure there is only one active controller manager.
      --metrics-bind-address string           The address the prometheus metric endpoint binds to. (default ":8080")
      --metrics-server-service-name string    Metrics server service name. Defaults to keda-metrics-apiserver (default "keda-metrics-apiserver")
      --metrics-service-bind-address string   The address the gRPRC Metrics Service endpoint binds to. (default ":9666")
      --operator-service-name string          Operator service name. Defaults to keda-operator (default "keda-operator")
      --validating-webhook-name string        ValidatingWebhookConfiguration name. Defaults to keda-admission (default "keda-admission")
      --webhooks-service-name string          Webhook service name. Defaults to keda-admission-webhooks (default "keda-admission-webhooks")
      --zap-devel                             Development Mode defaults(encoder=consoleEncoder,logLevel=Debug,stackTraceLevel=Warn). Production Mode defaults(encoder=jsonEncoder,logLevel=Info,stackTraceLevel=Error)
      --zap-encoder encoder                   Zap log encoding (one of 'json' or 'console') (default )
      --zap-log-level level                   Zap Level to configure the verbosity of logging. Can be one of 'debug', 'info', 'error', or any integer value > 0 which corresponds to custom debug levels of increasing verbosity (default )
      --zap-stacktrace-level level            Zap Level at and above which stacktraces are captured (one of 'info', 'error', 'panic').
      --zap-time-encoding time-encoding       Zap time encoding (one of 'epoch', 'millis', 'nano', 'iso8601', 'rfc3339' or 'rfc3339nano'). Defaults to 'epoch'. (default )
unknown flag: --k8s-cluster-name
```

## Steps to Reproduce the Problem

  1. Just install the Latest helm Chart with v2.12.1
  2. Observe the mismatch in Versions from offical release and Helm Values (required k8s args)

## Specifications

  - **KEDA Version:** 2.12.1
  - **Platform & Version:** Rancher Kubernets / Helm
  - **Kubernetes Version:** 1.X

**Comment by zroubalik** [comment]
@pickmylight are you sure you are using Chart 2.12.1? I downloaded the released tgz and this parameter is not present there. The feature using this paramter will be introduced in 2.13

**Comment by pickmylight** [comment]
Actually: There is no 2.12.1 release candidate. The current release is 2.12.0, which has not implemented this feature present in 2.13. The main question here is. When there is no such 2.12.1 release. Which branch of helm charts or release should be used when using the 2.12.1 release?

**Comment by zroubalik** [comment]
There is 2.12.1 Charts release, it is using KEDA 2.12.1 bits. This release doesn't include the mentioned parameter. Please double check that you are installing the correct version.

**Comment by pickmylight** [comment]
Just Checked. Yes, there is a version.
BUT it's only available via helm the Helm Charts Repo, and not as tagged release for usage with the 2.12.1 Keda artefacts. 
I will close this, as its clear, why I stumbled across the issue: for internal Air-Gapped Usage we use the Code and build the chart to push it into a self-hosted environment to use it on internal clusters. Since there was no Tag for 2.12.1  we used the main branch as it's the one with the most recent updates regarding the 2.12.1 release.
